WireAuthentication/WireAuthenticationAPI/Sources/Use cases/DetermineAuthenticationMethodUseCase.swift

```diff
@@ -0,0 +1,35 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+public protocol DetermineAuthenticationMethodUseCaseProtocol {
+
+    @MainActor
+    func invoke(emailOrSSOCode: String) async -> AuthenticationMethod
```

Comment:
I think it could be a good practice to always mark the invoke method as `@MainActor` because:

1. It creates a cleaner call side from the view model which should also be main actor isolated (i.e just create a `Task` and run the use case.
2. It makes it explicit to the implementation on which thread the use case is being called on and therefore the responsibility of the implantation to manage efficient concurrent execution.